RUFAS/data_structures/animal_to_manure_connection.py

```diff
@@ -50,11 +53,15 @@ class PenManureData:
     manure_urine_mass: float
     manure_urine_nitrogen: float
     stream_type: StreamType
+    first_processor: str = ""
```

Comment:
```suggestion
    first_processor: str
```
I don't think a default is necessary.

Reply:
When we're instantiating the first new PenManureData object in pen.py though we don't know the `first_processor` because it's the sum of all the PenManureData before it's divided up. That was why I had to set up the setter function to be able to assign it once the stream was divided and we could read the appropriate `first_processor` for that proportion of the stream with its PenManureData.

Reply:
Can we make this property `Optional` or `str | None`? Or if this would only add more complexity...